features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/SnapshotProducer.kt

```diff
@@ -54,35 +55,37 @@ internal class SnapshotProducer(
         parents: LinkedList<MobileSegment.Wireframe>,
         recordedDataQueueRefs: RecordedDataQueueRefs
     ): Node? {
-        val traversedTreeView = treeViewTraversal.traverse(view, mappingContext, recordedDataQueueRefs)
-        val nextTraversalStrategy = traversedTreeView.nextActionStrategy
-        val resolvedWireframes = traversedTreeView.mappedWireframes
-        if (nextTraversalStrategy == TraversalStrategy.STOP_AND_DROP_NODE) {
-            return null
-        }
-        if (nextTraversalStrategy == TraversalStrategy.STOP_AND_RETURN_NODE) {
-            return Node(wireframes = resolvedWireframes, parents = parents)
-        }
+        return withinBenchmarkSpan(view::class.java.simpleName) {
```

Comment:
can you please explain the approach? does it mean that effectively we will measure it only in the internal runs and for runs on the customer side it will be no-op and hence no performance penalty?

Reply:
Yes exactly, by default `GlobalBenchmark` has only ` NoOpBenchmarkProfiler` as the implementation, unless `GlobalBenchmark.register` is called to register a real implementation inside.

Currently the real implementation is in `:tools/benchmark` module, so only `:sample/benchmark` application who depends on it will have the real effect.